src/PhpSpreadsheet/ReferenceHelper.php

```diff
@@ -608,7 +608,12 @@ public function insertNewBefore($pBefore, $pNumCols, $pNumRows, Worksheet $pShee
         // Update workbook: define names
         if (count($pSheet->getParent()->getDefinedNames()) > 0) {
             foreach ($pSheet->getParent()->getDefinedNames() as $definedName) {
-                if ($definedName->getWorksheet()->getHashCode() === $pSheet->getHashCode()) {
+                $hash = null;
+                if ($definedName->getWorksheet()) {
```

Comment:
Just for the sake of readability, can you do an explicit comparison for not null)..

This is something that I've been doing in all the code that I've been working with recently, and it does make the if comparison a lot more explicit rather than relying on the loose typed truthiness/falsiness of PHP